llvm/lib/Transforms/Vectorize/VPlanTransforms.h

```diff
@@ -97,6 +97,13 @@ struct VPlanTransforms {
                                  unsigned BestUF,
                                  PredicatedScalarEvolution &PSE);
 
+  /// Try to simplify the branch condition of \p Plan. This may restrict the
+  /// resulting plan to \p BestVF and \p BestUF.
+  static bool simplifyBranchConditionForVFAndUF(VPlan &Plan,
+                                                ElementCount BestVF,
+                                                unsigned BestUF,
+                                                PredicatedScalarEvolution &PSE);
+
```

Comment:
This is only used in `optimizeForVFAndUF`, so it shouldn't need exposing here IIUC? It could be a static function in VPlanTransforms.cpp.

Reply:
Ah good point - done.